### PR TITLE
Add 2201.8.3 release note

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Add 2201.8.3 release note

### CODE REVIEW

**Review by sm1990 — APPROVED** [review]

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+layout: ballerina-left-nav-release-notes
+title: Swan Lake Update 8 (2201.8.3) 
+permalink: /downloads/swan-lake-release-notes/2201.8.3/
+active: 2201.8.3
+redirect_from:
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:6]:
```suggestion
redirect_from: 
```
Need to remove these redirections as they are not used anymore. Please check the template.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## Install Ballerina
+
+If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.
+
+## Runtime updates
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:33]:
```suggestion
To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).
```
This milestone link is empty.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+## Developer tools updates
+
+### Bug fixes
+To view bug fixes, see the GitHub milestone for Swan Lake Update 8 (2201.8.3) of the repositories below.
+
+- [Dev Tools](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3ATeam%2FDevTools+label%3AType%2FBug+is%3Aclosed)
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:40]:
```suggestion
- [Dev Tools](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3ATeam%2FDevTools+label%3AType%2FBug+is%3Aclosed)
```
This is also empty.

Comment (SasinduDilshara) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:40]: I think there is no difference with suggested changes

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md (hunk 2 above)
Comment (SasinduDilshara) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:33]: I think there is no difference with suggested changes

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md (hunk 3 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:40]: The link is empty.

Comment (SasinduDilshara) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:35]: It is because the issues are not closed since master PRs are not merged

Comment (SasinduDilshara) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:35]: Is it affect to the release note?

Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:35]: We can't have empty links on the release note. The release note should be released together with the release, which means, after completing the fixes and closing the issues.

Comment (SasinduDilshara) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:35]: We have to create a separate issues for 8.x then
I will work on that

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+## Install Ballerina
+
+If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.
+
+## Runtime updates
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).
+
```
Comment (lnash94) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:29]:
```suggestion

## Library updates
### Breaking changes

#### OpenAPI Tool
- Transitioned code generation from representing OpenAPI Specification byte format of string type as Ballerina `byte[]` to using Ballerina `string` type.

**OpenAPI Sample**
```Yaml
components:
  schemas:
       ByteExample:
	      type: string
              format: byte
```
**Old Generated Ballerina code**
```ballerina
type ByteExample byte[];
```
**New Generated Ballerina code**
```ballerina
type ByteExample string;
```
- Changed the client resource/remote code generation of `http:Response` to `nil` return type when the operation of OpenAPI specification has a response with no content.

**OpenAPI Sample**
```yaml
 /users/drafts/{id}:
    delete:
      operationId: deleteDraft
      parameters:
        - in: path
          name: id
          required: true
          schema:
            type: string
      responses:
        "200":
          description: Successful response //No content type
```
**Old Generated client resource/remote function**
```ballerina
    resource isolated function delete users/drafts/[string id]() returns http:Response|error {
        string resourcePath = string `/users/drafts/${getEncodedUri(id)}`;
        http:Response response = check self.clientEp-> delete(resourcePath);
        return response;
    }
```
**New Generated client resource/remote function**
```ballerina
resource isolated function delete users/drafts/[string id]() returns error? {
        string resourcePath = string `/users/drafts/${getEncodedUri(id)}`;
        return self.clientEp-> delete(resourcePath);
}
```
### Bug fixes

To view bug fixes, see the GitHub milestone for Swan Lake Update 2201.8.3 of the repositories below.
- [OpenAPI](https://github.com/ballerina-platform/ballerina-library/issues?q=is%3Aissue+label%3Amodule%2Fopenapi-tools+label%3AType%2FBug+milestone%3A2201.8.3+is%3Aclosed)
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    }
+```
+### Bug fixes
+
+To view bug fixes, see the GitHub milestone for Swan Lake Update 2201.8.3 of the repositories below.
+- [OpenAPI](https://github.com/ballerina-platform/ballerina-library/issues?q=is%3Aissue+label%3Amodule%2Fopenapi-tools+label%3AType%2FBug+milestone%3A2201.8.3+is%3Aclosed)
+
+## Developer tools updates
+
+### Observability Improvements
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]:
```suggestion
### Observability Improvements
```
Need to change the heading levels.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+### Improvements
+
+#### Observability
+
+##### Provide `response_errors_total_value` metric
+
+Ballerina observability now provides total number of errors in responses as an metric via `response_errors_total_value`.
+
+##### Let the user define `service name` in traces
```
Comment (SasinduDilshara) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:100]:
```suggestion
##### Define `service name` in traces
```

Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]:
```suggestion
##### Define the `service name` in traces
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+To view bug fixes, see the GitHub milestone for Swan Lake Update 2201.8.3 of the repositories below.
+- [OpenAPI](https://github.com/ballerina-platform/ballerina-library/issues?q=is%3Aissue+label%3Amodule%2Fopenapi-tools+label%3AType%2FBug+milestone%3A2201.8.3+is%3Aclosed)
+
+## Developer tools updates
+
+### Improvements
+
+#### Observability
+
+##### Provide `response_errors_total_value` metric
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]:
```suggestion
##### Provide the `response_errors_total_value` metric
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+Ballerina observability now provides total number of errors in responses as an metric via `response_errors_total_value`.
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]:
```suggestion
Ballerina observability now provides the total number of errors in responses as a metric via `response_errors_total_value`.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+##### Let the user define `service name` in traces
+
+User can now add a suffix to the service name shown in the tracer providers (Jaeger, Zipkin & New Relic) by passing an environment variable in the runtime as given below.
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]:
```suggestion
You can now add a suffix to the service name shown in the tracer providers (Jaeger, Zipkin, and New Relic) by passing an environment variable in the runtime as given below. 
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.
+
+## Runtime updates
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).
+
+## Library updates
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:30]:
```suggestion
## Library updates
```
need to change the headings.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+#### OpenAPI Tool
+- Code generation has transitioned from representing the OpenAPI Specification byte format of string type as the Ballerina `byte[]` to using Ballerina `string` type due to an issue with the previously generated code. This change will break the compatibility with the existing generated code.
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]:
```suggestion
- Code generation has transitioned from representing the OpenAPI Specification byte format of the string type as Ballerina `byte[]` to using Ballerina `string` type due to an issue with the previously generated code. This change will break the compatibility with the existing generated code.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+**OpenAPI Sample**
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:60]:
```suggestion
**OpenAPI sample**
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+```yaml
+components:
+ schemas:
+  ByteExample:
+   type: string
+   format: byte
+```
+**Old Generated Ballerina code**
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:68]:
```suggestion
**Old generated Ballerina code**
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+```ballerina
+type ByteExample byte[];
+```
+**New Generated Ballerina code**
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:72]:
```suggestion
**New generated Ballerina code**
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+```ballerina
+type ByteExample string;
+```
+- Changed the client resource/remote code generation of `http:Response` to a `nil` return type when the operation in the OpenAPI specification has a response with no content. With this improvement, the already generated code will be altered when regenerated with version the Ballerina 2201.8.3.
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]:
```suggestion
- Changed the `client resource/remote code` generation of the `http:Response` to a `nil` return type when the operation in the OpenAPI specification has a response with no content. With this improvement, the already generated code will be altered when regenerated with the Ballerina Swan Lake Update 8 (2201.8.3) version.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+
+**OpenAPI Sample**
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:78]:
```suggestion
**OpenAPI sample**
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+      parameters:
+        - in: query
+          name: userId
+          schema:
+            type: string    
+      responses:
+        "200":
+          description: Successful response //No content type
+```
+**Old Generated client resource/remote function**
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:92]:
```suggestion
**Old generated client resource/remote function**
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+```ballerina
+    resource isolated function delete user/draft(string? userId = ()) returns http:Response|error {
+        string resourcePath = string `/users/drafts/${getEncodedUri(id)}`;
+        map<anydata> queryParam = {"userId": userId};
+        resourcePath = resourcePath + check getPathForQueryParam(queryParam);
+        http:Response response = check self.clientEp-> delete(resourcePath);
+        return response;
+    }
+```
+**New Generated client resource/remote function**
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:102]:
```suggestion
**New generated client resource/remote function**
```

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:29]: Reviewed.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md (hunk 5 above)
Comment (NipunaMadhushan) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]: Addressed the suggestions

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md (hunk 6 above)
Comment (NipunaMadhushan) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]: Addressed the suggestions

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+To view bug fixes, see the GitHub milestone for Swan Lake Update 8 (2201.8.3) of the repositories below.
+
+- [Dev Tools](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3ATeam%2FDevTools+label%3AType%2FBug+is%3Aclosed)
+- [OpenAPI](https://github.com/ballerina-platform/ballerina-library/issues?q=is%3Aissue+label%3Amodule%2Fopenapi-tools+label%3AType%2FBug+milestone%3A2201.8.3+is%3Aclosed)
+
+## Backward-incompatible changes
+
+- OpenAPI tool client code generation has transitioned from representing the OpenAPI Specification byte format of the string type as Ballerina `byte[]` to using Ballerina `string` type due to an issue with the previously generated code. This change will break the compatibility with the existing generated code.
+
+**OpenAPI sample**
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:65]:
```suggestion
**OpenAPI sample**
```
These should be indented.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+```
+$ BAL_OBSERVE_SERVICE_NAME_SUFFIX=<suffix> bal run
+```
+
+By default, the service name will be the base path of the service or the display name of the service.
+
+### Bug fixes
+To view bug fixes, see the GitHub milestone for Swan Lake Update 8 (2201.8.3) of the repositories below.
+
+- [Dev Tools](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3ATeam%2FDevTools+label%3AType%2FBug+is%3Aclosed)
```
[the remark this answers is not in the document]
Comment (SasinduDilshara) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md:58]: open API one is a improvement, so it will not include in bugs

**Review by praneesha — APPROVED** [review]

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]: Fixed.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md (hunk 19 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]: Fixed.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+
+If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.
+
+## Runtime updates
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).
+
+## Ballerina library updates
```
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]: Fixed.

File: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md [hunk 22]
```diff
@@ -0,0 +1,112 @@
...
+
+##### Define `service name` in traces
+
+You can now add a suffix to the service name shown in the tracer providers ([Jaeger](https://www.jaegertracing.io/), [Zipkin](https://zipkin.io/), and [New Relic](https://newrelic.com/welcome-back)) by passing an environment variable in the runtime as given below. 
+
+```
+$ BAL_OBSERVE_SERVICE_NAME_SUFFIX=<suffix> bal run
+```
+
+By default, the service name will be the base path of the service or the display name of the service.
```
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md]: Fixed.

**Pull request merged**